### PR TITLE
Bug 1880068: Take StatefulSets, Jobs and CronJobs into account during image prune

### PR DESCRIPTION
**Pull request by ricardomaraschini**
This commit makes image pruner aware of StatefulSets, Jobs and CronJobs existing in the cluster.

### CODE REVIEW

File: pkg/cli/admin/prune/imageprune/prune_test.go [hunk 1]
```diff
@@ -77,6 +79,9 @@ func TestImagePruning(t *testing.T) {
 		deployments                          kappsv1.DeploymentList
 		dcs                                  appsv1.DeploymentConfigList
 		rss                                  kappsv1.ReplicaSetList
+		sss                                  kappsv1.StatefulSetList
```
Comment (soltysh) [inline pkg/cli/admin/prune/imageprune/prune_test.go:82]: upstream took `sset` as a short alias for StatefulSet, just fyi

File: pkg/helpers/image/test/util.go [hunk 2]
```diff
@@ -287,6 +289,79 @@ func DS(namespace, name string, containerImages ...string) kappsv1.DaemonSet {
...
+	}
+}
+
+// CronJob creates and returns a CronJob object.
+func CronJob(namespace, name string, containerImages ...string) batchv1beta1.CronJob {
+	return batchv1beta1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+			SelfLink:  "/apis/apps/v1/deployments/" + name,
```
Comment (soltysh) [inline pkg/helpers/image/test/util.go:305]: I know it's tests, but let's make it still accurate :wink:

File: pkg/helpers/image/test/util.go [hunk 3]
```diff
@@ -287,6 +289,79 @@ func DS(namespace, name string, containerImages ...string) kappsv1.DaemonSet {
...
+	}
+}
+
+// Job creates and returns a Job object.
+func Job(namespace, name string, containerImages ...string) batchv1.Job {
+	return batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+			SelfLink:  "/apis/apps/v1/deployments/" + name,
```
Comment (soltysh) [inline pkg/helpers/image/test/util.go:332]: Ditto

File: pkg/helpers/image/test/util.go [hunk 4]
```diff
@@ -287,6 +289,79 @@ func DS(namespace, name string, containerImages ...string) kappsv1.DaemonSet {
...
+	}
+}
+
+// StatefulSet creates and returns a StatefulSet object.
+func StatefulSet(namespace, name string, containerImages ...string) kappsv1.StatefulSet {
+	return kappsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+			SelfLink:  "/apis/apps/v1/deployments/" + name,
```
Comment (soltysh) [inline pkg/helpers/image/test/util.go:355]: Ditto.

**Review by soltysh — COMMENTED** [review]
This looks mostly good, what's left here?

File: pkg/cli/admin/prune/imageprune/prune_test.go (hunk 1 above)
Comment (ricardomaraschini) [reply pkg/cli/admin/prune/imageprune/prune_test.go:82]: Adjusted.

File: pkg/helpers/image/test/util.go (hunk 2 above)
Comment (ricardomaraschini) [reply pkg/helpers/image/test/util.go]: Addressed.

File: pkg/helpers/image/test/util.go (hunk 3 above)
Comment (ricardomaraschini) [reply pkg/helpers/image/test/util.go]: Addressed.

File: pkg/helpers/image/test/util.go (hunk 4 above)
Comment (ricardomaraschini) [reply pkg/helpers/image/test/util.go]: Addressed.

**Comment by dmage** [comment]
/assign @soltysh
for approval

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**